### PR TITLE
Feature/cs4360 195 edit functionality

### PR DESCRIPTION
**Pull request by jjohnson509**

### CODE REVIEW

**Review by GitNickProgramming — APPROVED** [review]

File: pb-crm-ui/src/app/api/api.service.ts [hunk 1]
```diff
@@ -45,8 +45,9 @@ export class ApiService {
   patch<T>(url: string, object: any, options: {}, handleError = ApiService.handleError): Observable<T> {
     if (object.hasOwnProperty('id')) {
       // Dont allow the update of an id with a PUT
-      const { id, ...model } = object
-      object = model
+      const { id, ...model } = object;
+      console.log('MODEL: ', model);
```
Comment (Newms226) [inline pb-crm-ui/src/app/api/api.service.ts:49]: Un-needed log to console, all requests & responses are logged by interceptor

File: pb-crm-ui/src/app/contact-book/contact-cache.service.ts [hunk 2]
```diff
@@ -44,12 +44,11 @@ export class ContactCacheService {
 
   updateContact(updated: ContactModel): Observable<ContactModel> {
     // Object construction with the spread operator
-    const {id, ...req} = updated;
-    return this.contactService.updateContact(req).pipe(map((contact: ContactModel) => {
-      const filtered = this.remove(id);
+    return this.contactService.updateContact(updated).pipe(map((contact: ContactModel) => {
+      console.log('UPDATED CONTACT: ', updated);
```
Comment (Newms226) [inline pb-crm-ui/src/app/contact-book/contact-cache.service.ts:48]: When it comes to the cache, do you think we should log all updates? Maybe something like

```Typescript
get contacts$() { 
     return this._contacts$().asObservable()
          .pipe(tap(updatedContacts => console.log("New contacts emit", updatedContacts)); 
}

```

File: pb-crm-ui/src/app/contact-book/contact.service.ts [hunk 3]
```diff
@@ -34,6 +34,7 @@ export class ContactService {
 
   updateContact(contact: UpdateContactRequest): Observable<UpdateContactResponse>{
       const url = '/api/contacts/' + contact.id;
+      console.log('URL ' + url);
```
Comment (Newms226) [inline pb-crm-ui/src/app/contact-book/contact.service.ts:37]: Gonna mess with this too.

**Review by Newms226 — COMMENTED** [review]
Looks good, I'm gonna make some changes to how we log in the contact cache. Look for another PR to your PR lol

**Pull request closed without merging**